tools/webpack/packages.js

```diff
@@ -0,0 +1,202 @@
+/**
+ * External dependencies
+ */
+const LiveReloadPlugin = require( 'webpack-livereload-plugin' );
+const CopyWebpackPlugin = require( 'copy-webpack-plugin' );
+const postcss = require( 'postcss' );
+
+const { join, basename } = require( 'path' );
+const { get } = require( 'lodash' );
+
+/**
+ * WordPress dependencies
+ */
+const CustomTemplatedPathPlugin = require( '@wordpress/custom-templated-path-webpack-plugin' );
+const LibraryExportDefaultPlugin = require( '@wordpress/library-export-default-webpack-plugin' );
+
+const baseDir = join( __dirname, '../../' );
+
+/**
+ * Given a string, returns a new string with dash separators converedd to
+ * camel-case equivalent. This is not as aggressive as `_.camelCase` in
+ * converting to uppercase, where Lodash will convert letters following
+ * numbers.
+ *
+ * @param {string} string Input dash-delimited string.
+ *
+ * @return {string} Camel-cased string.
+ */
+function camelCaseDash( string ) {
+	return string.replace(
+		/-([a-z])/g,
+		( match, letter ) => letter.toUpperCase()
+	);
+}
+
+module.exports = function( env = { environment: 'production', watch: false } ) {
+	const mode = env.environment;
+	const suffix = mode === 'production' ? '.min': '';
+
+	const gutenbergPackages = [
+		'api-fetch',
+		'a11y',
+		'autop',
+		'blob',
+		'blocks',
+//		'block-library', // Not on npm yet.
+		'block-serialization-default-parser',
+		'block-serialization-spec-parser',
+		'components',
+		'compose',
+		'core-data',
+		'data',
+		'date',
+		'deprecated',
+		'dom',
+		'dom-ready',
+//		'edit-post', // Not on npm yet.
+		'editor',
+		'element',
+//		'escape-html', // Not on npm yet.
+		'hooks',
+		'html-entities',
+		'i18n',
+		'is-shallow-equal',
+		'keycodes',
+//		'list-reusable-blocks', // Not on npm yet.
+		'nux',
+		'plugins',
+		'redux-routine',
+//		'rich-text', // Not on npm yet.
+		'shortcode',
+		'token-list',
+		'url',
+		'viewport',
+		'wordcount',
+	];
+
+	const packagesStyles = [
+		'nux',
+		'components',
+		'editor',
+		'edit-post',
+	];
+
+	const externals = {
+		react: 'React',
+		'react-dom': 'ReactDOM',
+		tinymce: 'tinymce',
+		moment: 'moment',
+		jquery: 'jQuery',
+		lodash: 'lodash',
+		'lodash-es': 'lodash',
+	};
+
+	gutenbergPackages.forEach( ( name ) => {
+		externals[ `@wordpress/${ name }` ] = {
+			this: [ 'wp', camelCaseDash( name ) ],
+		};
+	} );
+
+	const config = {
+		mode,
+
+		entry: gutenbergPackages.reduce( ( memo, packageName ) => {
+			const name = camelCaseDash( packageName );
+			memo[ name ] = join( baseDir, `node_modules/@wordpress/${ packageName }` );
+			return memo;
```

Comment:
Actually, this is pretty interesting as we could reference `node_modules/@wordpress/${ packageName }` also from Gutenberg itself. Just noting in case we would like to have shared code to rule them all :)